lib/index.js

```diff
@@ -1719,6 +1719,49 @@ Stream.prototype.done = function (f) {
 };
 
 /**
+ * Returns the result of a stream to a nodejs-style callback function.
+ *
+ * If the stream contains a single value, it will call `cb`
+ * with the single item emitted by the stream (if present).
+ * If the stream is empty, `cb` will be called without any arguments.
+ * If an error is encountered in the stream, this function will stop
+ * consumption and call `cb` with the error.
+ * If the stream contains more than one item, it will throw an error.
+ *
+ * @id toCallback
+ * @section Consumption
+ * @name Stream.toCallback(cb)
+ * @param {Function} cb - the callback to provide the error/result to
+ * @api public
+ *
+ * _([1, 2, 3, 4]).collect().toCallback(function (err, result) {
+ *     // parameter result will be [1,2,3,4]
+ *     // parameter err will be null
+ * });
+ */
+
+Stream.prototype.toCallback = function (cb) {
+    var encounteredError = false;
+    return this.stopOnError(function(err) {
+        encounteredError = true;
+        cb(err);
+    }).take(2).toArray(function (arr) {
+        if (encounteredError) {
+            return;
+        }
+        if (arr.length > 1) {
+            throw new Error('toCallback called on stream emitting multiple values');
```

Comment:
This should propagate the error to the callback so that it can be handled properly in the async case. So `cb(new Error(...))`.